### PR TITLE
SSCS-10053- Initial application setup and configuring functional Test

### PR DESCRIPTION
**Pull request by dholaramgurjar**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-10053

### Change description ###
Initial application structure setup
Port number configuration
Test functional and integration test configuration. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

**Review by dattipoeHMCTS — APPROVED** [review]

**Review by dattipoeHMCTS — APPROVED** [review]

**Review by TeaWithLucas — APPROVED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/sscs/config/SwaggerPublisherTest.java [hunk 1]
```diff
@@ -31,15 +24,15 @@
     @Test
     @SuppressWarnings("PMD.JUnitTestsShouldIncludeAssert")
     void generateDocs() throws Exception {
-        byte[] specs = mvc.perform(get("/v3/api-docs"))
+        /*byte[] specs = mvc.perform(get("/v3/api-docs"))
```
Comment (ss-evoco) [inline src/integrationTest/java/uk/gov/hmcts/reform/sscs/config/SwaggerPublisherTest.java:27]: Is this still needed ? If not, delete it.

**Review by ss-evoco — DISMISSED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/sscs/config/SwaggerPublisherTest.java (hunk 1 above)
Comment (dholaramgurjar) [reply src/integrationTest/java/uk/gov/hmcts/reform/sscs/config/SwaggerPublisherTest.java:27]: Yes it needed and It will cover in swagger api change

**Review by dattipoeHMCTS — APPROVED** [review]

File: build.gradle [hunk 2]
```diff
@@ -1,3 +1,12 @@
+buildscript {
+  repositories {
+    jcenter()
```
Comment (TeaWithLucas) [inline build.gradle:3]: Remove this, depreciated

File: bin/init.sh [hunk 3]
```diff
@@ -1,91 +0,0 @@
-#!/usr/bin/env bash
```
Comment (TeaWithLucas) [inline bin/init.sh:1]: Delete this file

File: build.gradle [hunk 4]
```diff
@@ -1,3 +1,12 @@
...
+  }
+  dependencies {
+    classpath("net.serenity-bdd:serenity-gradle-plugin:2.4.34")
```
Comment (TeaWithLucas) [inline build.gradle:6]: Update to version 3.1.20

File: build.gradle [hunk 5]
```diff
@@ -125,16 +145,38 @@ jacocoTestReport {
...
 sonarqube {
   properties {
-    property "sonar.projectName", "Reform :: spring-boot-template"
-    property "sonar.projectKey", "uk.gov.hmcts.reform:spring-boot-template"
+    property "sonar.projectName", "SSCS - Hearings Api"
+    property "sonar.projectKey", "SSCSHA"
     property "sonar.coverage.jacoco.xmlReportPaths", "${jacocoTestReport.reports.xml.destination.path}"
+    property "sonar.pitest.mode", "reuseReport"
+    property "sonar.pitest.reportsDirectory", "build/reports/pitest"
+    property "sonar.exclusions", "src/main/java/uk/gov/hmcts/reform/sscs/Application.java" +
```
Comment (TeaWithLucas) [inline build.gradle:162]: Change to
"src/main/java/uk/gov/hmcts/reform/sscs/HearingApiApplication.java"

File: build.gradle [hunk 6]
```diff
@@ -125,16 +145,38 @@ jacocoTestReport {
...
+      "src/main/java/uk/gov/hmcts/reform/sscs/config/**,"
   }
 }
+pitest {
+  junit5PluginVersion = '0.12'
+  targetClasses = ['uk.gov.hmcts.reform.sscs.*']
+  excludedClasses = [
+    'uk.gov.hmcts.reform.sscs.config.*',
+    'uk.gov.hmcts.reform.sscs.Application'
```
Comment (TeaWithLucas) [inline build.gradle:171]: Change to  uk.gov.hmcts.reform.sscs.HearingApiApplication

File: build.gradle [hunk 7]
```diff
@@ -164,37 +206,93 @@ repositories {
...
 
   implementation group: 'org.apache.logging.log4j', name: 'log4j-api', version: log4JVersion
   implementation group: 'org.apache.logging.log4j', name: 'log4j-to-slf4j', version: log4JVersion
 
+  implementation group: 'net.serenity-bdd', name: 'serenity-core', version: versions.serenity
+  implementation group: 'net.serenity-bdd', name: 'serenity-junit', version: versions.serenity
+  implementation group: 'net.serenity-bdd', name: 'serenity-rest-assured', version: versions.serenity
+  implementation group: 'net.serenity-bdd', name: 'serenity-spring', version: versions.serenity
+
+  //implementation group: 'uk.gov.hmcts.reform', name: 'service-auth-provider-client', version: '4.0.0'
```
Comment (TeaWithLucas) [inline build.gradle:252]: Replace with

implementation group: 'com.github.hmcts', name: 'service-auth-provider-java-client', version: '4.0.3'
Implementation group: 'com.github.hmcts', name: 'auth-checker-lib', version: '2.1.5'

File: build.gradle [hunk 8]
```diff
@@ -164,37 +206,93 @@ repositories {
...
+    }
+    implementation(group: 'org.bouncycastle', name: 'bcprov-jdk15on', version: '1.64') {
+      because 'CVE-2020-15522'
+    }
+  }
+
 }
 
-mainClassName = 'uk.gov.hmcts.reform.demo.Application'
+mainClassName = 'uk.gov.hmcts.reform.sscs.Application'
```
Comment (TeaWithLucas) [inline build.gradle:292]: Change to
uk.gov.hmcts.reform.sscs.HearingApiApplication

File: build.gradle [hunk 9]
```diff
@@ -205,7 +303,43 @@ bootJar {
 rootProject.tasks.named("processSmokeTestResources") {
   duplicatesStrategy = 'include'
 }
+rootProject.tasks.named("processFunctionalTestResources") {
+  duplicatesStrategy = 'include'
+}
+
+task fortifyScan(type: JavaExec)  {
+  main = "uk.gov.hmcts.fortifyclient.FortifyClientMainApp"
```
Comment (TeaWithLucas) [inline build.gradle:311]: Change main (depreciated) to mainClass

File: config/owasp/suppressions.xml [hunk 10]
```diff
@@ -15,4 +15,83 @@
...
+     file name: openhtmltopdf-jsoup-dom-converter-1.0.0.jar
+     ]]></notes>
+    <packageUrl regex="true">^pkg:maven/com\.openhtmltopdf/openhtmltopdf\-jsoup\-dom\-converter@.*$</packageUrl>
+    <cve>CVE-2015-6748</cve>
+    <cve>CVE-2021-37714</cve>
+  </suppress>
+
+  <suppress   until="2022-08-01">
+    <notes><![CDATA[
+   file name: bcprov-jdk15on-1.64.jar
```
Comment (TeaWithLucas) [inline config/owasp/suppressions.xml:92]: Needs explanation for suppression

File: src/main/resources/uk/gov/hmcts/reform/sscs/git.properties [hunk 11]
```diff
@@ -0,0 +1,2 @@
+git.commit.id=f285aff1063870fe206ff0a1dc97f87cb9dc02a6
```
Comment (TeaWithLucas) [inline src/main/resources/uk/gov/hmcts/reform/sscs/git.properties:1]: Add this file to git ignore and delete it

File: src/main/resources/application.yaml [hunk 12]
```diff
@@ -12,14 +19,23 @@ management:
...
+    #services: ${WA_S2S_AUTHORIZED_SERVICES:ccd,ccd_data,ccd_gw,ccd_ps}
+    services: ${SSCS_S2S_AUTHORIZED_SERVICES:ccd,ccd_data,ccd_gw,ccd_ps}
 
 #If you use a database then uncomment below lines and update db properties accordingly
 spring:
   config:
     import: "optional:configtree:/mnt/secrets/"
   application:
-    name: Spring Boot Template
+    name: SSCS HEARINGS API
```
Comment (TeaWithLucas) [inline src/main/resources/application.yaml:38]: No need for all caps
SSCS Hearings API

File: Jenkinsfile_CNP [hunk 13]
```diff
@@ -0,0 +1,40 @@
+#!groovy
+
+@Library("Infrastructure")
+
+def type = "java"
+def product = "sscs"
+def component = "sscs-hearings-api"
```
Comment (TeaWithLucas) [inline Jenkinsfile_CNP:7]: Change to
hearings-api

File: build.gradle [hunk 14]
```diff
@@ -164,37 +206,93 @@ repositories {
...
 
-mainClassName = 'uk.gov.hmcts.reform.demo.Application'
+mainClassName = 'uk.gov.hmcts.reform.sscs.Application'
 
 bootJar {
-  archiveFileName = "spring-boot-template.jar"
+  archiveFileName = "sscs-hearings-api.jar"
 
   manifest {
     attributes('Implementation-Version': project.version.toString())
```
Comment (TeaWithLucas) [inline build.gradle:298]: Change to
attributes('Implementation-Title': project.name, 'Implementation-Version': project.version.toString())

**Review by TeaWithLucas — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 14 above)
Comment (dholaramgurjar) [reply build.gradle]: Done

File: build.gradle (hunk 2 above)
Comment (dholaramgurjar) [reply build.gradle]: jcenter() function removed

File: build.gradle (hunk 4 above)
Comment (dholaramgurjar) [reply build.gradle]: version updated to 3.1.20

File: build.gradle (hunk 5 above)
Comment (dholaramgurjar) [reply build.gradle]: Application name changed

File: build.gradle (hunk 6 above)
Comment (dholaramgurjar) [reply build.gradle]: Done

File: build.gradle (hunk 7 above)
Comment (dholaramgurjar) [reply build.gradle]: Done.

File: build.gradle (hunk 8 above)
Comment (dholaramgurjar) [reply build.gradle]: Done

File: config/owasp/suppressions.xml (hunk 10 above)
Comment (dholaramgurjar) [reply config/owasp/suppressions.xml:93]: explanation added for suppression

File: Jenkinsfile_CNP (hunk 13 above)
Comment (dholaramgurjar) [reply Jenkinsfile_CNP]: Done

File: src/main/resources/application.yaml (hunk 12 above)
Comment (dholaramgurjar) [reply src/main/resources/application.yaml]: Done

File: build.gradle (hunk 9 above)
Comment (dholaramgurjar) [reply build.gradle]: done

**Review by TeaWithLucas — APPROVED** [review]

**Review by bensonlievoco — APPROVED** [review]

**Pull request merged**